core/arch/arm/mm/core_mmu_lpae.c

```diff
@@ -896,12 +896,24 @@ static unsigned int get_physical_addr_size_bits(void)
 	if (CFG_CORE_ARM64_PA_BITS <= 44)
 		return TCR_PS_BITS_16TB;
 
-	/* Physical address can't exceed 48 bits */
-	COMPILE_TIME_ASSERT(CFG_CORE_ARM64_PA_BITS <= 48);
 	/* CFG_CORE_ARM64_PA_BITS <= 48 */
 	return TCR_PS_BITS_256TB;
 }
 
+static unsigned int get_physical_addr_size_bits(void)
+{
+	const unsigned int size_bits = read_id_aa64mmfr0_el1() &
+				       ID_AA64MMFR0_EL1_PARANGE_MASK;
+	unsigned int b = 0;
+
+	if (IS_ENABLED(CFG_AUTO_MAX_PA_BITS))
+		return size_bits;
```

Comment:
`size_bits` is an index in `const uint8_t map[] = { 32, 36, 40, 42, 44, 48, 52, };`, not the size in bits.

Reply:
Yes and no. It's an index in that map, but it's also the `PARange` field in `ID_AA64MMFR0_EL1` and the `IPS` field in `TCR_EL1`.

Reply:
right, my bad. This si ok.